### DISCUSSION TITLE
Ruby Library

### DISCUSSION THREAD

**Discussion by ankane** [delta-io/delta-rs#3018] [Show and tell]
Hi, just wanted to share a new set of [Ruby bindings](https://github.com/ankane/delta-ruby) for Delta Lake :tada:

The code for the Python bindings was really easy to follow.

If you have any feedback, let me know or feel free to create an issue in the project. I'm planning to publish precompiled gems in the near future to speed up installation (like with [Ruby Polars](https://github.com/ankane/ruby-polars)).

---

One thing I think would be nice for the Rust library would be to either:

1. Raise an error for unsupported types (like unsigned integers) OR
2. Perform the data type conversions (like the Python library does)

(maybe there could be an option for which one)

Currently, the Rust library allows you to write some unsupported types, which can cause errors with future operations (like deletes).

**Comment by rtyler** [comment]
Very cool! I'll take a deeper look over the weekend. Are you at all interested in pulling this into the delta-incubator for more collaboration with other folks in the delta-rs ecosystem?

**Reply by ankane** [reply]
Thanks @rtyler! I'd prefer to keep it as a separate project for now.
